k8s/common/pod.go

```diff
@@ -101,9 +101,9 @@ func WaitForPodsToBeDeleted(client v1.CoreV1Interface, podsToDelete []corev1.Pod
 }
 
 // WaitForPodDeletion waits for given timeout for given pod to be deleted
-func WaitForPodDeletion(client v1.CoreV1Interface, name string, namespace string, timeout time.Duration) error {
+func WaitForPodDeletion(client v1.CoreV1Interface, uid types.UID, namespace string, timeout time.Duration) error {
 	t := func() (interface{}, bool, error) {
-		p, err := GetPodByName(client, name, namespace)
+		p, err := GetPodByUID(client, uid, namespace)
```

Comment:
Instead of listing all the pods, can we just use GetPodByName and then verify that either the pod is not returned or does not have the passed UID. This will reduce the unnecessarily heavy call of getting all the pods in the system.

You can change the input to this function to just the get pod object.

Reply:
Done, added a check for comparing UIDs